manifests/init.pp

```diff
@@ -1,60 +1,424 @@
+#   Sets override_homedir in the sssd section of the config file
+#   Default value: null
+#   Used in: templates/sssd.conf.erb
+#
+# @param pam_verbosity [Integer]
+#   Sets the value of pam_verbosity in the pam section of the config file.
+#   Default value: 1
+#   Used in: templates/service/pam.erb
+#
+# @param pam_id_timeout [Integer]
+#   Sets the value of pam_id_timeout in the pam section of the config file.
+#   Default value: 5
+#   Used in: templates/service/pam.erb
+#
+# @param pam_pwd_expiration_warning [Integer]
+#   Sets pam_pwd_expiration_warning in the pam section of the config file.
+#   Default value: 7
+#   Used in: templates/service/pam.erb
+#
+# @param pam_trusted_users [Optional[String]]
+#   Sets pam_trusted_users in the pam section of the config file.
+#   Default value: null
+#   Used in: templates/service/pam.erb
+#
+# @param pam_public_domains [Optional[String]]
+#   Sets pam_public_domains in the pam section of the config file.
+#   Default value: null
+#   Used in: templates/service/pam.erb
+#
+# @param reconnection_retries [Integer]
+#   Sets the value of reconnection_retries in the nss, pam, and sssd sections
+#   of the sssd config file.
+#   Default value: 3
+#   Used in: templates/service/pam.erb
+#            templates/service/nss.erb
+#            templates/sssd.conf.erb
+#
+# @param re_expression [Optional[String]]
+#   Sets re_expression in the domain and sssd sections of the config file.
+#   Default value: null
+#   Used in: templates/sssd.conf.erb
+#            templates/domain.erb
+#
+# @param services [Sssd::Services]
+#   Sets the value of services in the sssd section of the config file.
+#   Default value: ['nss','pam','ssh','sudo']
+#   Used in: templates/sssd.conf.erb
+#
+# @param ssh_hash_known_hosts [Boolean]
+#   Sets the value of ssh_hash_known_hosts in the ssh section of the config
+#   Default value: true
+#   Used in: templates/service/ssh.erb
+#
+# @param ssh_known_hosts_timeout [Optional[Integer]]
+#   Sets the value of sssh_known_hosts_timeout in the ssh section of config
+#   Default value: null
+#   Used in: templates/service/ssh.erb
+#
+# @param sudo_timed [Boolean]
+#   Sets the value of sudo_timed in the sudo section of the config file.
+#   Default value: false
+#   Used in: templates/service/sudo.erb
+#
+# @param try_inotify [Boolean]
+#   Sets the try_inotify field in the sssd section of the conf file
+#   Default value: true
+#   Used in: templates/service/ssh.conf.erb
+#
+# @param user [Optional[String]]
+#   Sets the value of the user field in the sssd section of the conf file.
+#   Default value: null
+#   Used in: templates/sssd.conf.erb
+#
+# @param user_attributes [Optional[String]]
+#   Sets the user_attributes field in the nss section of the conf file
+#   Default value: null
+#   Used in: templates/service/nss.erb
+#
+# @param vetoed_shells [Optional[String]]
+#   Sets the value of the vetoed_shells field in the nss section of sssd conf
```

Comment:
There are several params that I just put in boilerplate for, because I wanted to get the PR in and didn't want to go read through sssd man pages to see what the vetoed_shells setting does. Do you think it's worthwhile to go through and document options that are already documented in sssd's docs? I can if you think it would be useful.